### PR TITLE
test: add user-login tests

### PR DESCRIPTION
**Pull request by vindard**

### CODE REVIEW

File: test/bats/user-login.bats [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  clear_cache
+
+  start_server
+}
+
+teardown_file() {
+  stop_server
+}
+
+@test "user-login: login user" {
```
Comment (dolcalmi) [inline test/bats/user-login.bats:15]: shouldn't this be a helper? (or at least part of it)

**Review by dolcalmi — CHANGES_REQUESTED** [review]

File: test/bats/user-login.bats (hunk 1 above)
Comment (vindard) [reply test/bats/user-login.bats:15]: We already have a helper for it. This is to explicitly test it outside the helper in an isolated way.

Not sure if this approach is repetitive/redundant? I like that we're explicitly asserting this as a test condition.

Comment (dolcalmi) [reply test/bats/user-login.bats:29]: I think is not necessary.. if login is not working the other tests will fail but if you want to be explicit you can include tests for no happy paths

Comment (bodymindarts) [reply test/bats/user-login.bats:29]: Agree - let's not add this. It just increases the e2e runtime but doesn't assert anything unique.

**Pull request closed without merging**